toolkit/AThread.cpp

```diff
@@ -21,14 +21,16 @@ void AThread::ThreadMain() {
     std::mutex mtx;
     std::unique_lock<std::mutex> lck(mtx);
 
-    while (!(this->_is_stop)) {
-        if (!this->Main()) {
-            DARWIN_LOG_DEBUG("AThread::ThreadMain:: Error in main function, stopping the thread");
-            _is_stop = true;
-            break;
-        }
+    while (not this->_is_stop) {
         // Wait for notification or until timeout
         this->_cv.wait_for(lck, std::chrono::seconds(_interval));
+        if (not _is_stop) {
+            if (not this->Main()) {
+                DARWIN_LOG_DEBUG("AThread::ThreadMain:: Error in main function, stopping the thread");
+                _is_stop = true;
+                break;
+            }
+        }
```

Comment:
I see the logic is mostly hte same, I don't understand the need for a change in this method

Reply:
the change is in the position of the this->_cv.wait_for()
While we immediately started the thread before, now it waits one iteration of _interval before launching the thread